backend/cmd/api/loaders.go

```diff
@@ -30,15 +31,17 @@ func getPersonalizedLoaders(
 	collectionLoader *loaders.Loader[int, *common.Collection],
 	roles []string,
 	languagePreferences common.LanguagePreferences,
+	randomizedCursor *utils.RandomizedCursor,
 ) *common.PersonalizedLoaders {
 	pq := queries.PersonalizedQueries(roles, languagePreferences)
 
 	key := fmt.Sprintf(
-		"%s-%s-%s-%s",
+		"%s-%s-%s-%s-%s",
 		strings.Join(roles, "-"),
 		strings.Join(languagePreferences.PreferredAudioLanguages, "-"),
 		strings.Join(languagePreferences.PreferredSubtitlesLanguages, "-"),
 		fmt.Sprintf("%t", languagePreferences.ContentOnlyInPreferredLanguage),
+		fmt.Sprintf("%v", randomizedCursor.Seed),
```

Comment:
This should actually only happen if we have a collection that *is known* to be randomized...